web_src/js/index.js

```diff
@@ -2957,6 +2966,13 @@ function selectRange($list, $select, $from) {
   } else {
     $issue.attr('href', `${$issue.attr('href')}%23${$select.attr('rel')}`);
   }
+
+  const matchedPermalink = $copyPermalink.attr('href').match(/#L\d+$|#L\d+-L\d+$/);
+  if (matchedPermalink) {
+    $copyPermalink.attr('href', $copyPermalink.attr('href').replace($copyPermalink.attr('href').substr(matchedPermalink.index), `#${$select.attr('rel')}`));
+  } else {
+    $copyPermalink.attr('href', `${$copyPermalink.attr('href')}#${$select.attr('rel')}`);
+  }
```

Comment:
Can we merge these similar code into common functions?

```
const $refInIssue = $('a.ref-in-new-issue');
const $copyPermalink = $('a.copy-line-permalink');

// and we do not need to check $refInIssue.length here

updateRefInIssueLink = function(anchor) {
  ...
}
updateCopyPermalink = function(anchor) {
  ...
}

if ($from) {
   if (a != b) {
       updateRefInIssueLink(`#L${a}-L${b}`)
       updateCopyPermalink(`#L${a}-L${b}`)
       return;
   }
}

updateRefInIssueLink('#Lxx')
updateCopyPermalink('#Lxx')

```

Reply:
Changed in https://github.com/go-gitea/gitea/pull/17145/commits/58a6bc9dd2f6a5e68b33c53d58561386f42512a9